### PR TITLE
feat: add multiarch build

### PR DESCRIPTION
**Pull request by kalidax**
This job will

- Build and push an image to dockerhub.
- Build tofnd binary, gpg sign it, zip it, create sha256 hashes for both zip and binary, upload files to release and S3.
- For binary build the job will run on each combination of OS/arch defined in the matrix.
- For the container build it will run one time only on ubuntu-18.04/amd64

Tests have been done here : https://github.com/axelarnetwork/tofnd/actions/workflows/matrix-build.yaml

### CODE REVIEW

**Review by ggutoski — APPROVED** [review]
Approving as codeowner but @talalashraf or @erain9 should approve before merging.

**Review by sdaveas — APPROVED** [review]

File: .github/workflows/build-docker-image-and-binaries.yaml
Comment (talalashraf) [inline .github/workflows/build-docker-image-and-binaries.yaml:76]: will this install the latest version of rust ? do we lock versions @sdaveas ?

**Review by talalashraf — COMMENTED** [review]
looks good other than one comment. if we use a particular version of rust we should lock it.

**Pull request closed without merging**
